File: app/src/main/java/com/moment/eyepetizer/utils/DensityUtil.java
package com.moment.eyepetizer.utils;

import android.content.Context;

/**
 * Created by moment on 16/5/26.
 */
public class DensityUtil {
    public DensityUtil() {
    }

    public static int dip2px(Context context, float dpValue) {
        float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5F);
    }

    public static int px2dip(Context context, float pxValue) {
        float scale = context.getResources().getDisplayMetrics().density;
        return (int) (pxValue / scale + 0.5F);
    }
}

File: app/src/main/java/com/moment/eyepetizer/utils/DensityUtil.kt
package com.moment.eyepetizer.utils

import android.content.Context

/**
 * Created by moment on 16/5/26.
 */
class DensityUtil {
    companion object {

        fun dip2px(context: Context, dpValue: Float): Int {
            val scale = context.resources.displayMetrics.density
            return (dpValue * scale + 0.5f).toInt()
        }

        fun px2dip(context: Context, pxValue: Float): Int {
            val scale = context.resources.displayMetrics.density
            return (pxValue / scale + 0.5f).toInt()
        }
    }
}
